pytrickle/client.py

```diff
@@ -113,9 +153,55 @@ async def stop(self):
         
         # Send sentinel value to stop egress loop
         try:
-            self.output_queue.put_nowait(None)
-        except queue.Full:
+            await self.output_queue.put(None)
+        except asyncio.QueueFull:
             pass
+            
+        # Cancel drain task if running
+        if self._processor_drain_task and not self._processor_drain_task.done():
+            self._processor_drain_task.cancel()
+            try:
+                await self._processor_drain_task
+            except asyncio.CancelledError:
+                pass
+                
+        # Stop frame processor workers if running in queue mode
+        try:
+            if getattr(self.frame_processor, "queue_mode", False):
+                await self.frame_processor.stop_queue_workers()
+        except Exception as e:
+            logger.warning(f"Unable to stop frame processor workers: {e}")
+            
+        # Stop the frame processor if it has a stop method (for ComfyStream integration)
+        # Use stop() for model preservation during stream switches
+        try:
+            if hasattr(self.frame_processor, 'stop') and callable(self.frame_processor.stop):
+                await self.frame_processor.stop()
+                logger.info("Frame processor stopped (models preserved)")
+        except Exception as e:
+            logger.warning(f"Unable to stop frame processor: {e}")
+
+    def _reset_timing_state(self):
+        """Reset client-level timing state to prevent cross-stream timestamp conflicts."""
+        # Clear all timing-related state to ensure fresh start
+        self._stream_start_time = None
+        self._first_video_timestamp = None
+        self._first_audio_timestamp = None
+        self._timestamp_offset = 0.0
+        self._first_video_ts_sec = None
+        
+        # Clear any additional timing state that might exist
```

Comment:
We should know these exist right? Why would they not exist?

Reply:
Removed timing state reset as it wasn't needed. Turned out to be a workaround for the broken client/protocol state management fixed in https://github.com/livepeer/pytrickle/pull/13